File: solutions/java/EqualizeArray.java
public class EqualizeArray{
	static int equalizeArray(int[] arr) {
        Arrays.sort(arr);
        int lastEl = 0; // as the lower constraint is 1 we can set the first to 0
        int biggerCount = 0;
        int currentCount = 0;
        
        for(int i = 0; i < arr.length; i++){
            if(i == 0){
                lastEl = arr[i];
                currentCount = 1;
                continue;
            }
            
            if(arr[i] == lastEl){
                currentCount++;
                if(i == arr.length - 1 && biggerCount < currentCount){
                    biggerCount = currentCount;
                }
            }else{
                if(biggerCount < currentCount){
                    biggerCount = currentCount;
                }
                lastEl = arr[i];
                currentCount = 1;
            }
        }
        return arr.length - biggerCount;
    }
}

File: solutions/ruby/equalize_array.rb
def equalize_array arr
	arr.sort!
	last_el = 0 # as the lower constraint is 1 we can set the first to 0
	bigger_count = 0
	current_count = 0

	arr.each.with_index do |el, idx|
		if idx == 0
			last_el = el 
			current_count = 1
			next
		end
		if el == last_el
			current_count = current_count + 1
			if (idx == arr.length - 1 && bigger_count < current_count)
				bigger_count = current_count	
			end
		else
			if bigger_count < current_count
				bigger_count = current_count			
			end
			last_el = el
			current_count = 1
		end
	end
	arr.length - bigger_count
end